enhancements/cgu-precaching-additional-images.md

```diff
@@ -0,0 +1,308 @@
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: cluster-group-upgrade-overrides
+      namespace: default
+    data:
+      additionalWorkloadImages: |
+        quay.io/foobar/application1@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+        quay.io/foobar/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+        ...
+        quay.io/foobar/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+    ```
+
+2. The CR is validated (a function verifies if the clusters specified in the CR are part of a `ManagedCluster`
+object).
+
+3. If the validation succeeds, the pre-caching reconciliation commences. Note that all the sites are
+pre-cached concurrently.
+
+4. The pre-cache `spec` is then validated by checking if the managed policies exist. A valid `spec` will
+result in the following condition:
+    ```text
+    Type:    "PrecacheSpecValid",
+    Status:  True,
+    Reason:  "PrecacheSpecIsWellFormed",
+    Message: "Precaching spec is valid and consistent"
+    ```
+
+5. Pre-caching commences upon successful validation of the pre-cache `spec`. The following conditions are
+set while pre-caching is in-progress:
+    ```text
+    Type:    "PrecachingSuceeded",
+    Status:  False,
+    Reason:  "InProgress",
+    Message: "Precaching is required and not done"
+    ```
+
+6. After the pre-caching is successfully completed, the following condition is set:
+    ```text
+    Type:    "PrecachingSuceeded",
+    Status:  True,
+    Reason:  "PrecachingCompleted",
+    Message: "Precaching is completed for all clusters"
+    ```
+
+7. The remaining workflow for TALM continues as normal from here onwards.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+
+### Risks and Mitigations
+
+N/A
+
+### Benefits
+
+The benefits of the proposed approach are as follows:
+- The solution extends the current pre-caching TALM functionality in a unified approach.
+- The proposal integrates with TALM to control the rollout of policy (ConfigMap) to the managed
+clusters.
+- No additional operators need to be installed (apart from TALM) and thus compute requirements
+remain unchanged.
+
+### Drawbacks
+
+A drawback of the proposed approach is that TALM now accepts the additional workload of pre-caching
+user specific images.
+
+## Design Details
+
+### Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+> 1. This requires exposing previously private resources which contain sensitive
+     information.  Can we do this?
+
+### Test Plan
+
+N/A
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+M/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
```

Comment:
Maybe note that TALM pre-caching already works across N-2 versions. ie TALM version N can pre-cache on a cluster that is up to N-2.

Reply:
Will add this statement. Thank you!

Reply:
Done.